pyroute2/netlink/__init__.py

```diff
@@ -1007,12 +1007,15 @@ def encode(self):
 
         # handle the array case
         if self._nla_array:
+            header_type = 1
             for value in self.getvalue():
                 cell = type(self)(data=self.data,
                                   offset=offset,
                                   parent=self)
                 cell._nla_array = False
-                cell['header']['type'] = 1
+                cell['header']['type'] = header_type
```

Comment:
A good question, does it break other array usage cases. Give me some time to check it.